includes/forms/class-llms-forms.php

```diff
@@ -816,6 +847,91 @@ private function is_block_visible( $block ) {
 
 	}
 
+	/**
+	 * Determine if a block is visible in the list it's contained based on LifterLMS Visibility Settings
+	 *
+	 * Fall back on `$this->is_block_visible()` if empty `$block_list` is provided.
+	 *
+	 * @since [version]
+	 *
+	 * @param array   $block      Parsed block array.
+	 * @param array[] $block_list The list of WP Block array `$block` comes from.
+	 * @return bool
+	 */
+	public function is_block_visible_in_list( $block, $block_list ) {
+
+		if ( empty( $block_list ) ) {
+			return $this->is_block_visible( $block );
+		}
+
+		$path = $this->get_block_path( $block, $block_list );
+		foreach ( $path as $block ) {
+			if ( ! $this->is_block_visible( $block ) ) {
+				/* This filter is defined below. */
+				return apply_filters( 'llms_forms_is_block_visible', false, $block, $block_list );
+			}
+		}
+
+		/**
+		 * Filter whether or not the block is visible in the list of blocks it's contained.
+		 *
+		 * @since [version]
+		 *
+		 * @param bool    $visible    Whether or not the block is visible.
+		 * @param array   $block      Parsed block array.
+		 * @param array[] $block_list The list of WP Block array `$block` comes from.
+		 */
+		return apply_filters( 'llms_forms_is_block_visible', true, $block, $block_list );
+
+	}
+
+	/**
+	 * Returns a list of block parents plus the block itself in reverse order
+	 *
+	 * @since [version]
+	 *
+	 * @param array   $block      Parsed block array.
+	 * @param array[] $block_list The list of WP Block array `$block` comes from.
+	 * @param int     $iterations Stores the number of iterations.
+	 * @return bool
+	 */
+	private function get_block_path( $block, $block_list, $iterations = 0 ) {
+
+		foreach ( $block_list as $_block ) {
+
+			if ( $block === $_block ) {
+				return array( $block );
+			}
+
+			if ( ! empty( $_block['innerBlocks'] ) ) {
+				foreach ( $_block['innerBlocks'] as $inner_block ) {
+
+					// The inner block needs to be merged to the path.
+					$to_merge = array( $inner_block );
+
+					if ( $block === $inner_block ) { // Inner block is the one we're looking for.
+						$path     = array( $block );
+						$to_merge = array(); // Inner block equals the path, no need to merge it.
+					} else {
+						$path = $this->get_block_path( $block, array( $inner_block ), $iterations + 1 );
+					}
+
+					if ( $path ) {
+						// First iteration, append first block too.
+						if ( ! $iterations ) {
+							$to_merge[] = $_block;
+						}
+
+						// Merge.
+						return array_merge( $path, $to_merge );
+					}
+				}
+			}
+		}
+
+		return false;
```

Comment:
```suggestion
		return array();
```

I've found a way to break this (cause a fatal error attempting to run `false` into `foreach()`) by passing in a `$block_list` that does not contain `$block` which technically shouldn't happen unless custom code uses this incorrectly *but* if we return `array()` in favor of false it works exactly as expected and `is_block_visible_in_list()` will return false, which I think would be the semantically correct response: the block isn't in the list so it cannot logically be visible within the list...